plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,458 @@
+					if len(nline)+2 >= len(xprttcpFields) {
+						for i, t := range xprttcpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_tcp", fields, tags)
+					}
+				case "udp":
+					if len(nline)+2 >= len(xprtudpFields) {
+						for i, t := range xprtudpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_udp", fields, tags)
+					}
+				}
+			}
+		}
+
+		if (version == "3" && nfs3Ops[first]) || (version == "4" && nfs4Ops[first]) {
+			tags["operation"] = first
+			if len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var mount string
+	var version string
+	var export string
+	var skip bool
+	var nfs3Ops map[string]bool
+	var nfs4Ops map[string]bool
+
+	var nfs3Fields = []string{
+		"NULL",
+		"GETATTR",
+		"SETATTR",
+		"LOOKUP",
+		"ACCESS",
+		"READLINK",
+		"READ",
+		"WRITE",
+		"CREATE",
+		"MKDIR",
+		"SYMLINK",
+		"MKNOD",
+		"REMOVE",
+		"RMDIR",
+		"RENAME",
+		"LINK",
+		"READDIR",
+		"READDIRPLUS",
+		"FSSTAT",
+		"FSINFO",
+		"PATHCONF",
+		"COMMIT",
+	}
+
+	var nfs4Fields = []string{
+		"NULL",
+		"READ",
+		"WRITE",
+		"COMMIT",
+		"OPEN",
+		"OPEN_CONFIRM",
+		"OPEN_NOATTR",
+		"OPEN_DOWNGRADE",
+		"CLOSE",
+		"SETATTR",
+		"FSINFO",
+		"RENEW",
+		"SETCLIENTID",
+		"SETCLIENTID_CONFIRM",
+		"LOCK",
+		"LOCKT",
+		"LOCKU",
+		"ACCESS",
+		"GETATTR",
+		"LOOKUP",
+		"LOOKUP_ROOT",
+		"REMOVE",
+		"RENAME",
+		"LINK",
+		"SYMLINK",
+		"CREATE",
+		"PATHCONF",
+		"STATFS",
+		"READLINK",
+		"READDIR",
+		"SERVER_CAPS",
+		"DELEGRETURN",
+		"GETACL",
+		"SETACL",
+		"FS_LOCATIONS",
+		"RELEASE_LOCKOWNER",
+		"SECINFO",
+		"FSID_PRESENT",
+		"EXCHANGE_ID",
+		"CREATE_SESSION",
+		"DESTROY_SESSION",
+		"SEQUENCE",
+		"GET_LEASE_TIME",
+		"RECLAIM_COMPLETE",
+		"LAYOUTGET",
+		"GETDEVICEINFO",
+		"LAYOUTCOMMIT",
+		"LAYOUTRETURN",
+		"SECINFO_NO_NAME",
+		"TEST_STATEID",
+		"FREE_STATEID",
+		"GETDEVICELIST",
+		"BIND_CONN_TO_SESSION",
+		"DESTROY_CLIENTID",
+		"SEEK",
+		"ALLOCATE",
+		"DEALLOCATE",
+		"LAYOUTSTATS",
+		"CLONE",
+		"COPY",
+		"OFFLOAD_CANCEL",
+		"LOOKUPP",
+		"LAYOUTERROR",
+		"COPY_NOTIFY",
+		"GETXATTR",
+		"SETXATTR",
+		"LISTXATTRS",
+		"REMOVEXATTR",
+	}
+
+	nfs3Ops = make(map[string]bool)
+	nfs4Ops = make(map[string]bool)
+
+	if len(n.IncludeOperations) == 0 {
+		for _, Op := range nfs3Fields {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range nfs4Fields {
+			nfs4Ops[Op] = true
+		}
+	} else {
+		for _, Op := range n.IncludeOperations {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range n.IncludeOperations {
+			nfs4Ops[Op] = true
+		}
+	}
+
+	if len(n.ExcludeOperations) > 0 {
+		for _, Op := range n.ExcludeOperations {
+			if nfs3Ops[Op] {
+				delete(nfs3Ops, Op)
+			}
+			if nfs4Ops[Op] {
+				delete(nfs4Ops, Op)
+			}
+		}
+	}
```

Comment:
You can move this whole section into a `func (n *NFSClient) Init() error` function. It will be automatically called by telegraf _once_ after all config options are parsed and the corresponding fields are set. Then you only need to add `nfs3Ops` and `nfs4Ops` to the `NFSClient` structure to avoid doing this include/exclude in every cycle.

Reply:
Agreed.  Optimizing this bit has been on my todo list for a while.  It falls into the "get it correct first, optimize later" category.  Please see commit 996c4e4d.

Reply:
Please mark as resolved.